SAGE-1581: updated to leverage the cloud manifest

jswantek opened the pull request:
- apply a `zone` to any compute node that specifies one
- add a `resource.` label for all compute node's capabilities
- remove the hard-coded `gpu` labeling in favor of the node compatibility
- only apply a `resource.` label for a sensor if the manifest lists it
  and the hardware if physically detected

jswantek committed:
SAGE-1581: updated to leverage the cloud manifest

- apply a `zone` to any compute node that specifies one
- add a `resource.` label for all compute node's capabilities
- remove the hard-coded `gpu` labeling in favor of the node compatibility
- only apply a `resource.` label for a sensor if the manifest lists it
  and the hardware if physically detected

gemblerz commented on main.py:
```diff
@@ -44,64 +79,64 @@ def main():
+
+        # get the manifest compute dict for this node
+        for compute in manifest["computes"]:
+            if compute["serial_no"] in args.kubenode:
```
I would recommend lowering strings in case the serial_no contains any capital latter.

jswantek replied:
good idea. done

jswantek committed:
SAGE-1581: add automated CI simulating hardware tests

- tests full WSN (nxcore, nxagent, rpi-shield, rpi-enclosure)
- tests a simple Blade (sb-core only)
- tests missing hardware resource detection
- tests mis-match in manifest vs found compute devices
